{{cookiecutter.profile_name}}/OSLayer.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+import subprocess
+from typing import Tuple
+import random
+import string
+
+
+class OSLayer:
+    """
+    This class provides an abstract layer to communicating with the OS.
+    Its main purpose is to enable OS operations mocking, so we don't actually need to make file operations or create
+    processes.
+    """
+    @staticmethod
+    def mkdir(directory: Path):
+        directory.mkdir(parents=True, exist_ok=True)
+
+    @staticmethod
+    def remove_file(file: Path):
+        if file.exists():
+            file.unlink()
+
+    @staticmethod
+    def run_process_and_get_output_and_error_stream(cmd: str) -> Tuple[str, str]:
```

Comment:
This name is a bit too verbose (and leads to some of the tests/mocks being a bit unreadable). I would suggest renaming to `run_process`. If you want to be make it very clear that it returns stdout and stderr you could illustrate this by defining two types and the return typing will make things clear

```py
stdout = str
stderr = str
def run_process(cmd: str) -> Tuple[stdout, stderr]:
```

Reply:
I am not saying you *have* to define those types. Just thought I would mention them in case you were worried about readability. I think it is pretty clear that the function return stdout and stderr...

Reply:
You are right, and I really like the `Tuple[stdout, stderr]` return type annotation! Implemented in https://github.com/leoisl/snakemake-lsf/commit/31202c405b6f5c0aa38967c3064282ffad1a37f7